pytorch_lightning/callbacks/pruning.py

```diff
@@ -215,175 +217,183 @@ def _create_pruning_fn(self, pruning_fn: str, *args, **kwargs):
             pruning_fn = _PYTORCH_PRUNING_METHOD[pruning_fn]
             self._global_kwargs = kwargs
             return pruning_fn
-        else:
-            return ModelPruning._wrap_pruning_fn(_PYTORCH_PRUNING_FUNCTIONS[pruning_fn], **kwargs)
+        return ModelPruning._wrap_pruning_fn(_PYTORCH_PRUNING_FUNCTIONS[pruning_fn], **kwargs)
 
     @staticmethod
-    def _wrap_pruning_fn(pruning_fn, *args, **kwargs):
+    def _wrap_pruning_fn(pruning_fn, **kwargs):
         return partial(pruning_fn, **kwargs)
 
-    def _make_pruning_permanent(self):
+    def make_pruning_permanent(self):
+        """ Makes ``parameters_to_prune`` current pruning permanent. """
         for module, param_name in self._parameters_to_prune:
-            pytorch_prune.remove(module, param_name)
-
-    def _resolve_amount(self, current_epoch: int) -> float:
-        if isinstance(self.amount, Callable):
-            amount_fn = self.amount
-            amount = amount_fn(current_epoch)
-        else:
-            amount = self.amount
-        return amount
+            try:
```

Comment:
Not sure about try expect. We just to check for weight attribute.

Reply:
Unfortunately, PyTorch's `remove` does not have a `strict=False` argument and will raise a `ValueError` when not found. So it's the easiest way to check it.

https://github.com/pytorch/pytorch/blob/56034636b9f37b54dacfab8794e93ba6b10bdfc7/torch/nn/utils/prune.py#L1182-L1185